storage/namespace.go

```diff
@@ -696,7 +700,7 @@ func (n *dbNamespace) Flush(
 	}
 	n.RUnlock()
 
-	if !n.nopts.NeedsFlush() {
+	if !n.nopts.FlushEnabled() {
```

Comment:
unrelated to your changes, but while you're here - should `namespace.NeedsFlush()` check this too?

Reply:
If I was writing it from scratch I would say so, but I'd argue the way everything is currently structured the NeedsFlush() method is just used to see if the data has been flushed yet. A little hesitant to change it as part of this P.R

Reply:
fair enough. mind opening an issue to track

Reply:
done: https://github.com/m3db/m3db/issues/522